deploy/lib/deployFunctions.js

```diff
@@ -6,49 +6,47 @@ module.exports = {
   deployFunctions() {
     this.serverless.cli.log("Deploying Functions...");
     return BbPromise.bind(this)
-      .then(this.deployEachFunction)
-      .then(() =>
-        this.serverless.cli.log(
-          "Waiting for function deployments, this may take multiple minutes..."
-        )
-      )
-      .then(this.printFunctionInformationAfterDeployment);
+      .then(this.deployEachFunction);
   },
 
   deployEachFunction() {
-    const promises = this.functions.map((func) =>
-      this.deployFunction(func.id, {})
-    );
-
-    return Promise.all(promises);
+    return BbPromise.map(this.functions, (func) => {
+        return this.deployFunction(func.id, {})
+          .then((func) => {
+            this.serverless.cli.log(`Deploying ${func.name}...`)
+            return func
+          })
+          .then((func) => this.waitForFunctionStatus(func.id, "ready"))
+          .then((func) => this.printFunctionInformationAfterDeployment(func))
+          .then((func) => this.waitForDomainsDeployment(func))
+      }, {concurrency: 5})
   },
 
-  printFunctionInformationAfterDeployment() {
```

Comment:
For reviewers: this function was doing too many things:

- wait functions have been deployed
- print some info
- wait for domains to be deployed

I've split it in 3:

- wait functions have been deployed: `waitForFunctionStatus(func.id, "ready")`. This function already exist, I just had to fixed a mistake
- print some info: `printFunctionInformationAfterDeployment`
- wait for domains to be deployed: `waitForDomainsDeployment`

The benefit of using `deployFunction(...).then(waitForFunctionStatus(...))` is that now, we're waiting the function to be fully deployed. Before, we were just calling the `/deploy` endpoint and all function builds & deployments ran at the same time (and was throttled on server side).